pkg/knuu/instance_otel.go

```diff
@@ -0,0 +1,356 @@
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (i *Instance) createJaegerReceiver() Jaeger {
+	return Jaeger{
+		Protocols: JaegerProtocols{
+			GRPC: JaegerGRPC{Endpoint: fmt.Sprintf("localhost:%d", i.obsyConfig.jaegerGrpcPort)},
+			ThriftCompact: JaegerThriftCompact{
+				Endpoint: fmt.Sprintf("localhost:%d", i.obsyConfig.jaegerThriftCompactPort),
+			},
+			ThriftHTTP: JaegerThriftHTTP{
+				Endpoint: fmt.Sprintf("localhost:%d", i.obsyConfig.jaegerThriftHttpPort),
+			},
+		},
+	}
+}
+
+func (i *Instance) createReceivers() Receivers {
+	receivers := Receivers{}
+
+	if i.obsyConfig.otlpPort != 0 {
+		receivers.OTLP = i.createOtlpReceiver()
+	}
+
+	if i.obsyConfig.prometheusPort != 0 {
+		receivers.Prometheus = i.createPrometheusReceiver()
+	}
+
+	if i.obsyConfig.jaegerGrpcPort != 0 {
+		receivers.Jaeger = i.createJaegerReceiver()
+	}
+
+	return receivers
+}
+
+func (i *Instance) createOtlpHttpExporter() OTLPHTTPExporter {
+	return OTLPHTTPExporter{
+		Auth: OTLPAuth{
+			Authenticator: "basicauth/otlp",
+		},
+		Endpoint: i.obsyConfig.otlpEndpoint,
+	}
+}
+
+func (i *Instance) createJaegerExporter() JaegerExporter {
+	return JaegerExporter{
+		Endpoint: i.obsyConfig.jaegerEndpoint,
+		TLS: TLS{
+			Insecure: true,
+		},
+	}
+}
+
+func (i *Instance) createExporters() Exporters {
+	exporters := Exporters{}
+
+	if i.obsyConfig.otlpEndpoint != "" {
+		exporters.OTLPHTTP = i.createOtlpHttpExporter()
+	}
+
+	if i.obsyConfig.jaegerEndpoint != "" {
+		exporters.Jaeger = i.createJaegerExporter()
+	}
+
+	return exporters
+}
+
+func (i *Instance) createService() Service {
+	var extensions []string
+	if i.obsyConfig.otlpEndpoint != "" {
+		extensions = append(extensions, "basicauth/otlp")
+	}
+
+	pipelines := Pipelines{}
+	if i.obsyConfig.otlpPort != 0 || i.obsyConfig.prometheusPort != 0 || i.obsyConfig.otlpEndpoint != "" {
+		metrics := Metrics{}
+		if i.obsyConfig.otlpPort != 0 {
+			metrics.Receivers = append(metrics.Receivers, "otlp")
+		}
+		if i.obsyConfig.prometheusPort != 0 {
+			metrics.Receivers = append(metrics.Receivers, "prometheus")
+		}
+		if i.obsyConfig.otlpEndpoint != "" {
+			metrics.Exporters = append(metrics.Exporters, "otlphttp")
+		}
+		pipelines.Metrics = metrics
+	}
+	if i.obsyConfig.jaegerGrpcPort != 0 || i.obsyConfig.jaegerThriftCompactPort != 0 || i.obsyConfig.jaegerThriftHttpPort != 0 || i.obsyConfig.jaegerEndpoint != "" {
+		traces := Traces{}
+		if i.obsyConfig.otlpPort != 0 {
+			traces.Receivers = append(traces.Receivers, "otlp")
+		}
+		if i.obsyConfig.jaegerGrpcPort != 0 || i.obsyConfig.jaegerThriftCompactPort != 0 || i.obsyConfig.jaegerThriftHttpPort != 0 {
+			traces.Receivers = append(traces.Receivers, "jaeger")
+		}
+		if i.obsyConfig.otlpEndpoint != "" {
+			traces.Exporters = append(traces.Exporters, "otlphttp")
+		}
+		if i.obsyConfig.jaegerEndpoint != "" {
+			traces.Exporters = append(traces.Exporters, "jaeger")
+		}
+		pipelines.Traces = traces
+	}
```

Comment:
Let's refactor this into prepare metrics and traces private funcs. 
You can make also make an extra-mile either now or in next PRs where we can squash those if statements + appends